src/Service/HourReportService.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace App\Service;
+
+use App\Exception\EconomicsException;
+use App\Model\Reports\HourReportData;
+use App\Model\Reports\HourReportProjectTag;
+use App\Model\Reports\HourReportProjectTicket;
+use App\Model\Reports\HourReportTimesheet;
+use App\Repository\IssueRepository;
+use App\Repository\ProjectRepository;
+use App\Repository\VersionRepository;
+use App\Repository\WorklogRepository;
+
+class HourReportService
+{
+    public function __construct(
+        private readonly IssueRepository $issueRepository,
+        private readonly WorklogRepository $worklogRepository,
+        private readonly ProjectRepository $projectRepository,
+        private readonly VersionRepository $versionRepository,
+    ) {
+    }
+
+    /**
+     * @throws EconomicsException
+     */
+    public function getProjects(): array
+    {
+        $projects = $this->projectRepository->findAll();
+        $projectChoices = [];
+        foreach ($projects as $project) {
+            $projectChoices[$project->getName()] = $project->getId();
+        }
+
+        return $projectChoices;
+    }
+
+    public function getMilestones(string $projectId, bool $includeAllOption = false): array
+    {
+        $milestones = $this->versionRepository->findBy(['project' => $projectId]);
+        $milestoneChoices = [];
+        if ($includeAllOption) {
+            $milestoneChoices['All milestones'] = '0';
+        }
+        foreach ($milestones as $milestone) {
+            $milestoneName = $milestone->getName() ?? '';
+            $milestoneChoices[$milestoneName] = $milestone->getId();
+        }
+
+        return $milestoneChoices;
+    }
+
+    /**
+     * @throws EconomicsException
+     */
+    public function getHourReport(string $projectId, ?\DateTimeInterface $fromDate, ?\DateTimeInterface $toDate, int $versionId = null): HourReportData
```

Comment:
We have some potential performance issues to consider with this code.
If we have many issues in a project, each with many worklogs, we have to potentially load many entities in memory. 
The handling of issues should perhaps be paged, so the entity manager can be cleared during calculations.

Have you tested with many issues, maybe with fixutres?